content/300-guides/050-database/870-using-prisma-with-mongodb.mdx

```diff
@@ -0,0 +1,351 @@
+
+## How to implement Prisma with MongoDB
+
+This section provides instructions for how to carry out tasks that require steps specific to MongoDB.
+
+### How to migrate old data to new schemas
+
+Migrating your database over time is an important part of the development cycle. During development, you will need to update your schema file (for example, toads new fields), then update the data in your your development environment’s database, and eventually push both the updated schema and the new data to the production database.
+
+<Admonition type="info">
+
+When using MongoDB, be aware that the “coupling” between your schema and the database is purposefully designed to be less rigid than with with SQL databases; MongoDB will not enforce the schema, so you have to verify data integrity.
+
+</Admonition>
+
+These iterative tasks of updating the schema and the database can result in inconsistencies between your schema and the actual data in the database. Let’s look at one scenario where this can happen, and then examine several strategies for you and your team to consider for handling these inconsistencies.
+
+**Scenario**: you need to add a new field to your schema to accommodate a billing address that is separate from the currently stored shipping address. You currently have the following schema in your `schema.prisma` file:
+
+```prisma file=prisma/schema.prisma
+model Product {
+  id     String  @id @default(auto()) @map("_id") @db.ObjectId
+  name   String  @unique
+}
+
+model Order {
+  id              String   @id @default(auto()) @map("_id") @db.ObjectId
+  product         Product  @relation(fields: [productId], references: [id])
+  shippingAddress Address
+  productId       String   @db.ObjectId
+}
+
+type Address {
+  street String @id @default(auto()) @map("_id") @db.ObjectId
+  city   String
+  zip    String
+}
+```
+
+There are a number of strategies you could use for migrating this schema:
+
+- **"On-demand" updates**: with this strategy, you and your team have agreed that updates can be made to the schema as needed. However, in order to avoid migration failures due to inconsistencies between the data and schema, there is agreement in the team that any new fields added are explicitly defined as optional.
+
+  In our scenario above, you can go ahead and add the new `billingAddress` field to the `User` model in your Prisma schema, and define it as optional:
```

Comment:
In line with simplification, on demand migrations could also apply to a single field. We could say that you find out you want to add a phoneNumber to the user model. You'd add it as optional, then as users fill it in from their profile, it gets updated. This might be slightly simpler to explain than using a composite.

Reply:
I've updated the example to use a phone number instead of a billing address, and tried to clarify what 'on demand' means using your comment here https://github.com/prisma/docs/pull/3017#discussion_r840940479